src/vs/code/electron-main/windows.ts

```diff
@@ -730,6 +745,12 @@ export class WindowsManager implements IWindowsService {
 		// Emit events
 		iPathsToOpen.forEach(iPath => this.eventEmitter.emit(EventTypes.OPEN, iPath));
 
+		// Add to backups
```

Comment:
The intent here is for workspaces.json to contain a key for each workspace currently opened in order to support the crash recovery use case, so it's being added on the main process side when the window is launched.

This is also used [to determine whether a single window is opened](https://github.com/Microsoft/vscode/pull/12400/files#diff-56386098163653fba6a65ac4e37ea24bR123) (there is probably another way to do this) in order to decide whether the backups can be discarded safely, but that's not the main reason for it.